### PR TITLE
add fortran language

### PR DESCRIPTION
**Pull request by mcbunkus**
This PR adds syntax highlighting and LSP support for Fortran. It doesn't include textobjects but `fortls` seems to work without any issues.

Let me know if I made any mistakes, or if there's anything I need to change. 😄

### CODE REVIEW

File: runtime/queries/fortran/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,170 @@
+(identifier) @variable
+(string_literal) @string
+(number_literal) @number
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:3]:
```suggestion
(number_literal) @constant.numeric
```

File: runtime/queries/fortran/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+(boolean_literal) @boolean
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:4]:
```suggestion
(boolean_literal) @constant.builtin.boolean
```

File: runtime/queries/fortran/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,170 @@
+(identifier) @variable
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:1]: It looks like these queries were written for nvim-treesitter. nvim-treesitter has reversed precedence so this has the lowest priority for nvim but everything else that uses tree-sitter uses the reverse: this is the highest precedence stanza in helix, so the `@variable.parameter` capture on L136 can't match. To fix it you can move this line down below any other line that captures the `(identifier)` node.

File: runtime/queries/fortran/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+[
+  "if" 
+  "then"
+  "else"
+  "elseif"
+  "endif"
+  "where"
+  "endwhere"
+] @conditional
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:85]:
```suggestion
] @keyword.control.conditional
```

File: runtime/queries/fortran/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+[
+  "do"
+  "enddo"
+  "while"
+  "forall"
+] @repeat
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:92]:
```suggestion
] @keyword.control.repeat
```

File: runtime/queries/fortran/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  (name) @function)
+
+(subroutine_call
+	(name) @function)
+
+(keyword_argument
+  name: (identifier) @keyword)
+
+(derived_type_member_expression
+  (type_member) @property)
```
Comment (the-mikedavis) [inline runtime/queries/fortran/highlights.scm:170]:
```suggestion
  (type_member) @variable.other.member)
```

**Comment by mcbunkus** [comment]
I've fixed the queries, thanks for the help. I'm a complete noob to treesitter.

**Review by the-mikedavis — APPROVED** [review]
Looks great, thanks!

**Comment by David-Else** [comment]
@mcbunkus Could you add install instructions at https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers ?

**Pull request merged**